### PR TITLE
PhpUnitConstructFixer - Fix more use cases

### PR DESCRIPTION
**Pull request by SpacePossum**
Fix the following (Not) cases as well.

```php
<?php $this->assertNotSame(false, $a);
<?php $this->assertNotFalse($a);
```

```php
<?php $this->assertNotSame(true, $a);
<?php $this->assertNotTrue($a);
```

```php
<?php $this->assertNotSame(false, $a);
<?php $this->assertNotFalse($a);
```

```php
<?php $this->assertNotSame(true, $a);
<?php $this->assertNotTrue($a);
```
Replaces https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1380 (which was targeted at master and more of a mess).

### CODE REVIEW

**Comment by SpacePossum** [comment]
fabbot tries to fix the test cases, this is not an issue IMO.

**Comment by keradus** [comment]
Yes, fabbot is broken about that
cc @fabpot

File: Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php [hunk 1]
```diff
@@ -23,76 +23,14 @@ class PhpUnitConstructFixerTest extends AbstractFixerTestBase
      */
     public function testFix($expected, $input = null)
     {
-        $fixer = $this->getFixer();
-
-        $fixer->configure(array(
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php]: why is this removed?

**Comment by fabpot** [comment]
@keradus I would be more than happy to remove this check whenever php-cs-fixer does that for me :)

**Comment by keradus** [comment]
@fabpot I think this fixer do what you want (see https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.11/Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php#L87-L90)
(but only on 1.11@dev)

File: Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php (hunk 1 above)
Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php]: merge gone wrong from changes of weeks ago... I'll revert sorry I missed this one before opening the PR

Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php]: restored and squashed

File: Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php [hunk 2]
```diff
@@ -106,5 +78,42 @@ public function provideTestFixCases()
     );',
             ),
         );
+
+        $types = array('true', 'false', 'null');
+        // Equals' => true, 'NotEquals' => false); are transformed by PhpUnitStrictFixer to 'Same' and 'NotSame'
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php]: but this fixer can do it too, so why not test it?

Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php]: Good point, I cannot recall why I removed those from the initial PR.
Update the PR with `Equals` and I've simplified the test code a bit.

File: Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php [hunk 3]
```diff
@@ -106,5 +78,27 @@ public function provideTestFixCases()
...
         );
+
+        return array_merge(
+            $cases,
+            $this->generateCases('<?php $this->assert%s(%s, $a, "%s", "%s");', '<?php $this->assert%s%s($a, "%s", "%s");'),
+            $this->generateCases('<?php $this->assert%s(%s, $a); //%s %s', '<?php $this->assert%s%s($a); //%s %s')
+        );
+    }
+
+    private function generateCases($fromTemplate, $toTemplate)
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php]: everywhere we use expected, input order.
it would be great to be consistent ;)

**Comment by keradus** [comment]
:+1:

File: Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php (hunk 3 above)
Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php]: fair enough, done and squashed :)

**Comment by SpacePossum** [comment]
>@fabpot I think this fixer do what you want 

which is the same fixer being adjusted in this PR ^-^

File: Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php (hunk 3 above)
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php]: That is why I don't like squashing when it is not needed.
sth goes wrong:
travis-ci.org/FriendsOfPHP/PHP-CS-Fixer/jobs/85576570#L200

Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php]: I forgot to swap the inputs as well, things should be ok now :|

**Comment by keradus** [comment]
Thank you @SpacePossum.
